File: app/src/main/java/alejandromr92/com/movies/presentation/presenter/GetPopularMoviesPresenter.java
package alejandromr92.com.movies.presentation.presenter;

import alejandromr92.com.movies.presentation.model.MovieView;

import java.util.List;

public interface GetPopularMoviesPresenter {

    void getPopularMovies(int page);

    interface View {
        void onPopularMoviesRetrieved(List<MovieView> movieList);
        void onPopularMoviesRetrievingError(int errorCode);
    }
}

File: app/src/main/java/alejandromr92/com/movies/presentation/presenter/GetPopularMoviesPresenter.kt
package alejandromr92.com.movies.presentation.presenter

import alejandromr92.com.movies.presentation.model.MovieView

interface GetPopularMoviesPresenter {
    fun getPopularMovies(page: Int)

    interface View : BaseView {
        fun onPopularMoviesRetrieved(movieList: List<MovieView>)
        fun onPopularMoviesRetrievingError()
    }
}